calblueprint/meepanyar
# Refactor UserData and pull filtered user info with sites

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
This PR updates UserData to match the redux structure of CustomerData, SiteData, and InventoryData using selectors and entity adapters. With this change, we no longer use `mapStateToProps` anywhere in the app, as we've completely transitioned over to using selectors!

Additionally, this PR pulls basic user info (id, username, name, admin, siteIds) for all users at a given site. The user data is filtered in the backend so only essential details are passed through.

## How to review
Test with `annie/user-data` on the backend.

- Use the Redux inspector to make sure the data looks right.
- Navigate to inventory and select a purchase request to see that the username shows up. If the user doesn't have a username, it will fall back to the user's id
- Feedback on naming conventions

## Relevant Links

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs
Backend PR: https://github.com/calblueprint/meepanyar-node/pull/26

Based on the refactoring work from:
SiteData #83 
CustomerData #71 
InventoryData #60 

## Next steps
- Integrate with settings

### Screenshots
n/a

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
